autoload/codefmt/formatterhelpers.vim

```diff
@@ -62,3 +66,40 @@ function! codefmt#formatterhelpers#AttemptFakeRangeFormatting(
 
   call maktaba#buffer#Overwrite(1, line('$'), l:full_formatted)
 endfunction
+
+
+""
+" @public
+" Resolve a flag (function, string or array) to a normalized array, with special
+" handling to convert a spaceless string to a single-element array. This is the
+" common case for executables, and more importantly, is backward-compatible for
+" existing user settings.
+"
+" @throws WrongType if the flag doesn't resolve to a string or array
+function! codefmt#formatterhelpers#ResolveFlagToArray(flag_name) abort
+  let l:FlagFn = s:plugin.Flag(a:flag_name)
+  if maktaba#value#IsFuncref(l:FlagFn)
+    let l:flag = maktaba#function#Call(l:FlagFn)
+  else
+    let l:flag = l:FlagFn
+  endif
+  if type(l:flag) is# type([])
+    return l:flag
+  elseif type(l:flag) is# type("") && l:flag !~ '\s'
+    return l:flag == "" ? [] : [l:flag]
+  endif
+  throw maktaba#error#WrongType(
+        \ a:flag_name . ' flag must be array after calling. Found %s',
```

Comment:
This single error message doesn't clearly capture all the cases. For instance, it's not accurate to say if you pass a string containing a space that it "must be array after calling". I'd suggest splitting out a couple different error messages for different cases.

Reply:
done in #138